### PR TITLE
Cleaner Git version check

### PR DESCRIPTION
**Pull request by kevgo**

### CODE REVIEW

**Comment by kevgo** [comment]
The `git` package just provides connectivity to Git. It can provide the currently installed version. It doesn't have opinions about whether the version is too low for Git Town.

File: src/cmd/root.go [hunk 1]
```diff
@@ -25,10 +25,15 @@ and it allows you to perform many common Git operations faster and easier.`,
 
 // Execute runs the Cobra stack.
 func Execute() {
-	if err := git.CheckVersion(); err != nil {
+	majorVersion, minorVersion, err := git.Version()
+	if err != nil {
 		fmt.Println(err)
 		os.Exit(1)
 	}
+	if majorVersion*100+minorVersion < 207 {
```
Comment (charlierudolph) [inline src/cmd/root.go:33]: This feels weird to me. Thoughts on `majorVersion > 2 || (majorVersion == 2 and minorVersion > 7)`?

Comment (kevgo) [reply src/cmd/root.go:33]: Great idea, that's what I was looking for. Implemented in #1547. I'm checking a bunch of edge cases (2.6, 3.0, etc) using unit tests so that we still only have one end-to-end test.
